ibm/service/codeengine/resource_ibm_code_engine_function.go

```diff
@@ -0,0 +1,845 @@
+	return &schema.Resource{
+		CreateContext: resourceIbmCodeEngineFunctionCreate,
+		ReadContext:   resourceIbmCodeEngineFunctionRead,
+		UpdateContext: resourceIbmCodeEngineFunctionUpdate,
+		DeleteContext: resourceIbmCodeEngineFunctionDelete,
+		Importer:      &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "project_id"),
+				Description:  "The ID of the project.",
+			},
+			"code_binary": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Specifies whether the code is binary or not. Defaults to false when `code_reference` is set to a data URL. When `code_reference` is set to a code bundle URL, this field is always true.",
+			},
+			"code_main": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "main",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "code_main"),
+				Description:  "Specifies the name of the function that should be invoked.",
+			},
+			"code_reference": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "code_reference"),
+				Description:  "Specifies either a reference to a code bundle or the source code itself. To specify the source code, use the data URL scheme and include the source code as base64 encoded. The data URL scheme is defined in [RFC 2397](https://tools.ietf.org/html/rfc2397).",
+			},
+			"code_secret": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "code_secret"),
+				Description:  "The name of the secret that is used to access the specified `code_reference`. The secret is used to authenticate with a non-public endpoint that is specified as`code_reference`.",
+			},
+			"managed_domain_mappings": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "local_public",
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "managed_domain_mappings"),
+				Description:  "Optional value controlling which of the system managed domain mappings will be setup for the function. Valid values are 'local_public', 'local_private' and 'local'. Visibility can only be 'local_private' if the project supports function private visibility.",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_code_engine_function", "name"),
+				Description:  "The name of the function.",
+			},
+			"run_env_variables": {
+				Type:        schema.TypeList,
```

Comment:
Can we use [TypeSet ](https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typeset).
With TypeList when the order changes in user tf and in state file it always show a diff when more than one element is provided